libcodechecker/server/client_db_access_handler.py

```diff
@@ -101,43 +120,113 @@ def __init__(self,
         self.report_ident = sqlalchemy.orm.query.Bundle('report_ident',
                                                         Report.id,
                                                         Report.bug_id,
-                                                        Report.run_id,
-                                                        Report.start_bugevent,
-                                                        Report.start_bugpoint)
+                                                        Report.run_id)
+
+    def __lastBugEventPos(self, report_id):
+        """
+        This function returns the last BugPathEvent object position which
+        belongs to the given report. If no such event is found then None
+        returns.
+        """
+        last = self.__session.query(BugPathEvent) \
+            .filter(BugPathEvent.report_id == report_id) \
+            .order_by(BugPathEvent.order.desc()) \
+            .limit(1).all()
+
+        if len(last) < 1:
+            return None
+
+        last = last[0]
+
+        bpe = bugpathevent_db_to_api(last)
+        bpe.filePath = self.__session.query(File).get(bpe.fileId).filepath
+
+        return bpe
+
+    def __sortResultsQuery(self, query, sort_types=None):
+        """
+        Helper method for __queryDiffResults and queryResults to apply sorting.
+        """
+
+        # Get a list of sort_types which will be a nested ORDER BY.
+        sort_type_map = {SortType.FILENAME: [File.filepath],
+                         SortType.CHECKER_NAME: [Report.checker_id],
+                         SortType.SEVERITY: [Report.severity]}
+
+        # Mapping the SQLAlchemy functions.
+        order_type_map = {Order.ASC: asc, Order.DESC: desc}
+
+        if sort_types is None:
+            sort_types = [SortMode(SortType.FILENAME, Order.ASC)]
+
+        for sort in sort_types:
+            sorttypes = sort_type_map.get(sort.type)
+            for sorttype in sorttypes:
+                order_type = order_type_map.get(sort.ord)
+                query = query.order_by(order_type(sorttype))
+
+        return query
+
+    @timeit
+    def getRunData(self, run_name_filter):
+
+        session = self.__session
+        results = []
+        try:
+            # Count the reports subquery.
+            stmt = session.query(Report.run_id,
+                                 func.count(literal_column('*')).label(
+                                     'report_count')) \
+                .filter(Report.suppressed == false()) \
+                .group_by(Report.run_id) \
```

Comment:
I don't really understand this question. Doesn't this code generate a count() function in SQL? By the way this is not my code. The diff indicates this function new because it has been moved in the source code from another place.